src/eval/merge.rs

```diff
@@ -527,13 +535,16 @@ fn rev_thunks<'a, I: Iterator<Item = &'a mut RichTerm>>(map: I, env: &mut Enviro
 pub mod hashmap {
     use std::collections::HashMap;
 
+    pub struct SplitResult<K, V1, V2> {
+        pub left: HashMap<K, V1>,
+        pub center: HashMap<K, (V1, V2)>,
+        pub right: HashMap<K, V2>,
+    }
+
     /// Split two hashmaps m1 and m2 in three parts (left,center,right), where left holds bindings
     /// `(key,value)` where key is not in `m2.keys()`, right is the dual (keys of m2 that are not
     /// in m1), and center holds bindings for keys that are both in m1 and m2.
-    pub fn split<K, V1, V2>(
-        m1: HashMap<K, V1>,
-        m2: HashMap<K, V2>,
-    ) -> (HashMap<K, V1>, HashMap<K, (V1, V2)>, HashMap<K, V2>)
+    pub fn split<K, V1, V2>(m1: HashMap<K, V1>, m2: HashMap<K, V2>) -> SplitResult<K, V1, V2>
```

Comment:
Clippy was complaining that the return type was too verbose, and both @yannham and I agreed with Clippy here, so we extracted the return type in a `struct`, which makes the code clearer in our opinion